hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMDirectoryCreateRequest.java

```diff
@@ -121,6 +122,18 @@ public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
 
   }
 
+  public OmLockOpr.OmLockInfo lock(OzoneManager ozoneManager, OmLockOpr lockOpr) throws IOException {
```

Comment:
Why are we touching FSO/Legacy requests in this OBS patch?

Reply:
@errose28 We have, `OMDirectoryCreateRequest for OBS` and OMDirectoryCreateRequestWithFSO for FSO.
So above class is OBS only (not FSO)
Legacy many places is treated as OBS only, so from this code perspective, it can not distinguish if LEGACY or OBS, so precautionary added the check for OBS.